src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests.cs

```diff
@@ -1951,5 +1951,36 @@ public void TestWithUsingStatic2()
         {
             TestMissing(@"using [|Sample|] ; ");
         }
+
+        [WorkItem(1107929)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
+        public void TestAccesiblityForPublicFields()
+        {
+            Test(
+@"class A { public B b = new [|B|](); }",
+@"public class B { public B() { } }",
+index: 0,
+isAddedDocument:true);
+        }
+
+        [WorkItem(1107929)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
+        public void TestAccesiblityForPublicFields2()
+        {
+            Test(
+@"class A { public B b = new [|B|](); }",
+@"class A { public B b = new B(); } public class B { public B() {}}",
+index: 1);
+        }
+
+        [WorkItem(1107929)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
+        public void TestAccesiblityForPublicFields3()
+        {
+            Test(
+@"class A { public B b = new [|B|](); }",
+@"class A { public B b = new B(); public class B { public B() {}}}",
+index: 2);
+        }
```

Comment:
Could you also test the cases where ```B``` is fully qualified / alias qualified? Would we generate in the correct location in such cases?